docs/source/parameters.rst

```diff
@@ -0,0 +1,373 @@
+   ----------- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+    SIZE        10   10   10   20   20   20   30   30   30   40   40   40
+   ----------- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+    ITER        10   20   30   10   20   30   10   20   30   10   20   30
+   =========== ==== ==== ==== ==== ==== ==== ==== ==== ==== ==== ==== ====
+
+Notice that using the pgen input method makes it trivially easy to add 1000's of parameters, something which would be cumbersome via manual editing of the global.parameters block in the study specification file.
+
+There are no requirements to cram all of the logic into the **get_custom_generator** function.  The next example demonstrates using 3rd party libraries and breaking out the actual parameter generation algorithm into separate helper functions that the **get_custom_generator** function uses to get some more complicated distributions.  The only concerns with this approach will be to ensure the library is installed in the same virtual environment as the maestro executable you are using.  The simple parameter distribution demoed in here is for single variables and is often encountered in polynomial interpolation applications and is designed to suppress the Runge phenomena by sampling the function to be interpolated at the Chebyshev nodes.
+
+EXAMPLE:
+  Using ``numpy`` to calculate a sampling of a function at the Chebyshev nodes.
+
+.. literalinclude:: ../../samples/parameterization/np_cheb_pgen_pargs.py
+   :language: python
+   :caption: np_cheb_pgen_pargs.py
+   :linenos:
+
+
+Running this parameter generator with the following pargs
+
+.. code-block:: bash
+
+   $ maestro run study.yaml --pgen np_cheb_pgen.py --parg "X_MIN:0" --parg "X_MAX:3" --parg "NUM_PTS:11"
+
+results in the 1D distribution of points for the **X** parameter shown by the orange circles:
+
+.. image:: pgen_images/cheb_map.png
+
+
+Referencing Values from a Specification's Env Block
+===================================================
+
+In addition to command line arguments via :ref:`pargs_section`, the variables defined in the **env** block in the workflow specification file can be accessed inside the :py:class:`~maestrowf.datastructures.core.ParameterGenerator` objects, which is passed in to **get_custom_generator** as the first argument.  The lulesh sample specification can be extended to store the default values for the **pgen**, enhancing the reproducability of the generator.  The following example makes use of an optional **seed** parameter which can be added to the **env** block or set via **pargs** to make a repeatable study specification, while omitting it can enable fully randomized workflows upon every instantiation.  The variables are accessed via the :py:class:`~maestrowf.datastructures.core.StudyEnvironment`'s :py:func:`~maestrowf.datastructures.core.StudyEnvironment.find()` function, which will return ``None`` if the variable is not defined in the study specification.
```

Comment:
I think when referring to sections the ** bolding works just fine. @jwhite242 -- what do you think?

Reply:
If you mean the :ref:`pargs_section`, that's a link, not just formatting.  And following from the previous comment, is the suggestion here that any key/block name that can be in a YAML specification should be formatted as ` ``item`` ` instead of ``**item**``?, so seed and env would get the former, and pgen, pargs, get_custom_generator would be the latter? (mixed feelings on the last one not being formatted as code, but could go either way)

Reply:
I think that `get_custom_generator` and `seed` look better. **pargs** is a command line option, so I think that could go either way. Out of curiosity, is **pgen** bolded to match **pargs**? -- I think those could be put into tick marks (I think that would match the style elsewhere).